src/static/index.js

```diff
@@ -304,25 +226,72 @@ function noMatchFallback(url) {
   app.ports.onPageRequested.send(url);
 }
 
-function getAstarteClient(session) {
-  if (!session || session == "null") {
+function getAstarteClient(config) {
+  if (!config || config == "null") {
     return null;
   }
 
-  const config = JSON.parse(session).api_config;
-  const protocol = config.secure_connection ? "https://" : "http://";
-  const astarteConfig = {
-    realm: config.realm,
-    token: config.token,
-    secureConnection: config.secure_connection,
-    realmManagementUrl: protocol + config.realm_management_url,
-    appengineUrl: protocol + config.appengine_url,
-    pairingUrl: protocol + config.pairing_url,
-    flowUrl: protocol + config.flow_url,
-    enableFlowPreview: config.enable_flow_preview,
+  // base API URL
+  const astarteApiUrl = config.astarte_api_url;
+  let appEngineApiUrl = "";
+  let realmManagementApiUrl = "";
+  let pairingApiUrl = "";
+  let flowApiUrl = "";
+
+  if (astarteApiUrl === "localhost") {
+    appEngineApiUrl = new URL("http://localhost:4002");
+    realmManagementApiUrl = new URL("http://localhost:4000");
+    pairingApiUrl = new URL("http://localhost:4003");
+    flowApiUrl = new URL("http://localhost:4009");
+
+  } else if (typeof astarteApiUrl === "string") {
+    appEngineApiUrl = new URL("appengine/", astarteApiUrl);
+    realmManagementApiUrl = new URL("realmmanagement/", astarteApiUrl);
+    pairingApiUrl = new URL("pairing/", astarteApiUrl);
+    flowApiUrl = new URL("flow/", astarteApiUrl);
```

Comment:
Are the trailing `/` needed? Let's make sure we don't have double slashes as we currently do

Reply:
Those are required to build URLs that can be used later as base URLs. Without them, the URL refers to a resource called for example "flow"

examples:
```
foo = new URL("foo", "https://www.example.com")
// https://www.example.com/foo

relative_bar = new URL("bar", foo);
// https://www.example.com/bar

absolute_bar = new URL("/bar", foo);
// https://www.example.com/bar
```

with trailing slashes
```
foo = new URL("foo/", "https://www.example.com")
// https://www.example.com/foo/

relative_bar = new URL("bar", foo);
// https://www.example.com/foo/bar

absolute_bar = new URL("/bar", foo);
// https://www.example.com/bar
```